elasticsearch/run.sh

```diff
@@ -60,4 +69,72 @@ else
 	exit 1
 fi
 
+
+# Wait for Elasticsearch port to be opened. Fail on timeout or if response from Elasticsearch is unexpected.
+wait_for_port_open() {
+	rm -f ${LOG_FILE}
+	echo -n "Checking if Elasticsearch is ready on ${ES_REST_BASEURL} "
+	while ! curl -s -k \
+			--cert ${secret_dir}admin-cert \
+			--key  ${secret_dir}admin-key \
+			--max-time ${max_time} \
+			-o ${LOG_FILE} \
+			${ES_REST_BASEURL} && [ ${timeouted} == false ]
+	do
+		echo -n "."
+		sleep ${RETRY_INTERVAL}
+		(( retry -= 1 ))
+		if (( retry == 0 )) ; then
+			timeouted=true
+		fi
+	done
+
+	# Test for response code 200 in Elasticsearch output. This can be sensitive to Elasticsearch version.
+	if [ -f ${LOG_FILE} ] && grep -q "200" ${LOG_FILE} ; then
+		echo "- connection successful"
+	else
+		if [ ${timeouted} == true ] ; then
+			echo -n "[timeout] "
+		fi
+		echo "failed"
+		cat ${LOG_FILE}
+		exit 1
+	fi
+}
+
+verify_or_add_index_templates() {
+	wait_for_port_open
+	# Try to wait for cluster become more stable before index template being pushed in.
+	# Give up on timeout and continue...
+	curl -v -s -k -X GET \
+		--cert ${secret_dir}admin-cert \
+		--key  ${secret_dir}admin-key \
+		"${ES_REST_BASEURL}/_cluster/health?wait_for_status=yellow&timeout=${max_time}s"
```

Comment:
I was trying to get more information about this situation here https://discuss.elastic.co/t/creating-index-template-anytime-regardless-cluster-health-status/56194 (it is bit confusing).
But let's think about this... if the cluster is red due to index data (recovering indices or the like) then there is a chance the index template is already there (may be upgrade scenario is exception). If the cluster is red and there is no data/indices in it then we are probably in trouble anyway.

This process is still far from ideal. I am for example not sure if we can currently avoid data being indexed into ES cluster before the index template is in place. In which case the index template will apply only to indices created bit later (next day?).

I think the issue here is that we try to handle initialization at individual node level, however, we in fact should init cluster only after it has been fully formed. It would be great if we had mechanism of blocking indexing into cluster until we can make sure things are correctly initialized. Not sure if this is possible now.